packages/api/src/lib/structures/Route.ts

```diff
@@ -1,15 +1,47 @@
 import { Piece } from '@sapphire/pieces';
 import type { Awaitable } from '@sapphire/utilities';
-import { Collection } from 'discord.js';
 import { RouteData } from '../utils/RouteData';
-import type { MethodCallback, RouteStore } from './RouteStore';
 import type { ApiRequest } from './api/ApiRequest';
 import type { ApiResponse } from './api/ApiResponse';
-import { methodEntries, type Methods } from './http/HttpMethods';
+import type { MethodName } from './http/HttpMethods';
 import type { MimeTypeWithoutParameters } from './http/Server';
 
 /**
  * @since 1.0.0
+ *
+ * @example A simple GET route that returns a JSON response:
+ * ```typescript
+ * // hello.get.ts
+ * import { Route } from '@sapphire/plugin-api';
+ *
+ * export class MyRoute extends Route {
+ *   public run(request: Route.Request, response: Route.Response) {
+ *     return response.json({ message: 'Hello, World!' });
+ *   }
+ * }
+ * ```
+ *
+ * ```bash
+ * $ curl http://localhost:4000/hello
+ * {"message":"Hello, World!"}
+ * ```
+ *
+ * @example A simple POST route that reads the body and returns it:
+ * ```typescript
+ * // echo.post.ts
+ * import { Route } from '@sapphire/plugin-api';
+ *
+ * export class MyRoute extends Route {
+ *   public run(request: Route.Request, response: Route.Response) {
+ *     return response.json(request.params);
+ *   }
+ * }
+ * ```
+ *
+ * ```bash
+ * $ curl -H "Content-Type: application/json" -d '{"hello":"world"}' http://localhost:4000/echo
```

Comment:
```suggestion
 * $ curl -X POST -H "Content-Type: application/json" -d '{"hello":"world"}' http://localhost:4000/echo
```

While curl is smart enough to make it a POST when a body is provided, it's still good to provide the request method if it's not GET. Especially because for PATCH and PUT there is no way for curl to automatically set the request method.